src/components/swap/SwapHeader.tsx

```diff
@@ -13,14 +22,136 @@ const StyledSwapHeader = styled.div`
+  display: flex;
+  flex-direction: row;
+  cursor: ${({ isClickable }) => isClickable && 'pointer'};
+  justify-content: center;
+  align-items: center;
+`
+
+const TOKEN_SAFETY_ARTICLE =
+  'https://support.uniswap.org/hc/en-us/articles/11306664890381-Why-isn-t-MoonPay-available-in-my-region-'
+
+// TODO(lynnshaoyu): add analytics and logging
+// TODO(lynnshaoyu): add animation & timeout to buy fiat unavailable toast
 export default function SwapHeader({ allowedSlippage }: { allowedSlippage: Percent }) {
+  const theme = useTheme()
+  const { account } = useWeb3React()
+  const openFiatOnRampModal = useOpenModal(ApplicationModal.FIAT_ONRAMP)
+  const fiatOnRampButtonEnabled = useFiatOnRampButtonEnabled()
+
+  const [buyFiatClicked, setBuyFiatClicked] = useBuyFiatClicked()
+  const [continueBuyFiatFlowPostRegionCheck, setContinueBuyFiatFlowPostRegionCheck] = useState(false)
+  const [continueBuyFiatFlowPostWalletModal, setContinueBuyFiatFlowPostWalletModal] = useState(false)
+
+  const [checkFiatRegionAvailability, setCheckFiatRegionAvailability] = useState(false)
+  const {
+    available: fiatOnrampAvailable,
+    availabilityChecked: fiatOnrampAvailabilityChecked,
+    error,
+    loading: fiatOnrampAvailabilityLoading,
+  } = useFiatOnrampAvailability(checkFiatRegionAvailability)
+
+  const disableBuyCryptoButton = Boolean(
+    error || (!fiatOnrampAvailable && fiatOnrampAvailabilityChecked) || fiatOnrampAvailabilityLoading
+  )
+
+  const [walletDrawerOpen, toggleWalletDrawer] = useWalletDrawer()
+
+  const handleBuyCrypto = useCallback(() => {
+    if (!fiatOnrampAvailabilityChecked) {
+      setCheckFiatRegionAvailability(true)
+      setContinueBuyFiatFlowPostRegionCheck(true)
+    } else if (fiatOnrampAvailable && !account && !walletDrawerOpen) {
+      toggleWalletDrawer()
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(true)
+    } else if (fiatOnrampAvailable && account) {
+      openFiatOnRampModal()
+      setBuyFiatClicked(true)
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(false)
+    } else if (!fiatOnrampAvailable) {
+      setBuyFiatClicked(true)
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(false)
+    }
+  }, [
+    walletDrawerOpen,
+    fiatOnrampAvailabilityChecked,
+    fiatOnrampAvailable,
+    account,
+    toggleWalletDrawer,
+    openFiatOnRampModal,
+    setBuyFiatClicked,
+  ])
+
+  useEffect(() => {
+    if (continueBuyFiatFlowPostRegionCheck || (account && continueBuyFiatFlowPostWalletModal)) {
+      handleBuyCrypto()
+    }
+  }, [
+    openFiatOnRampModal,
+    account,
+    checkFiatRegionAvailability,
+    fiatOnrampAvailabilityChecked,
+    fiatOnrampAvailable,
+    setBuyFiatClicked,
+    toggleWalletDrawer,
+    handleBuyCrypto,
+    setContinueBuyFiatFlowPostRegionCheck,
+    continueBuyFiatFlowPostWalletModal,
+    continueBuyFiatFlowPostRegionCheck,
```

Comment:
do you need all these dependencies in this array? seems like only 4 variables are used

Reply:
you are right yikes! i just didn't change cuz the linter wasn't complaining. thanks for pointing out.